network/peer.go

```diff
@@ -601,12 +591,8 @@ func (p *peer) handleGetVersion(_ message.InboundMessage) {
 
 // assumes the [stateLock] is not held
 func (p *peer) handleVersion(msg message.InboundMessage) {
-
-	// We always allow the first version message through; if we have already processed
-	// the first version message, we let the message through only if the version
-	// reset was requested, but the second version message not yet processed.
 	switch {
-	case p.gotVersion.GetValue() && (!p.resetVersion.GetValue() || p.regotVersion.GetValue()):
```

Comment:
Why this change? It seems like they had a reason to do this specific check in this condition but based on the context I have at the time of writing this comment, I don't see why the changes of the PR would make this obsolete

Reply:
I implemented this myself. This code allowed nodes to request a version reset once, which was needed because of a weird unnecessary check in the Avalanche networking code. On every `pong` we receive as an answer to a `ping`, the node checks if the version is compatible, and will disconnect if not. This check is strange, because in practice, Avalanche nodes should _never_ change their version unless they restart/disconnect.

However, this broke our code, where nodes started sending new version information after one hard fork, and then started rejecting the old version information after the next hard fork, leading to a disconnect for everyone on hard fork. To minimize disruption to operators, I implemented this fix before we deployed on Songbird, and no one got disconnected.

Now, as everyone is on a new stable version, that will not change except with code upgrades, it's no longer needed.